GVFS/GVFS.Service/GVFSService.Mac.cs

```diff
@@ -16,13 +20,32 @@ public class GVFSService
         private Thread serviceThread;
         private ManualResetEvent serviceStopped;
         private string serviceName;
+        private RepoRegistry repoRegistry;
+        private string serviceDataLocation;
+        private RequestHandler requestHandler;
 
         public GVFSService(JsonTracer tracer)
         {
-            this.tracer = tracer;
+            if (tracer != null)
+            {
+                string logFilePath = Path.Combine(
+                    GVFSPlatform.Instance.GetDataRootForGVFSComponent(GVFSConstants.Service.ServiceName),
+                    GVFSConstants.Service.LogDirectory);
+                Directory.CreateDirectory(logFilePath);
+
+                this.tracer = tracer;
+                this.tracer.AddLogFileEventListener(
+                    GVFSEnlistment.GetNewGVFSLogFileName(logFilePath, GVFSConstants.LogFileTypes.Service),
+                    EventLevel.Verbose,
+                    Keywords.Any);
+            }
+
             this.serviceName = GVFSConstants.Service.ServiceName;
             this.serviceStopped = new ManualResetEvent(false);
             this.serviceThread = new Thread(this.ServiceThreadMain);
+            this.serviceDataLocation = GVFSPlatform.Instance.GetDataRootForGVFSComponent(this.serviceName);
+            this.repoRegistry = new RepoRegistry(this.tracer, new PhysicalFileSystem(), this.serviceDataLocation);
+            this.requestHandler = new RequestHandler(this.tracer, EtwArea, this.repoRegistry);

```

Comment:
In `GVFSService.Windows.cs` `this.repoRegistry` and `this.requestHandler` are created in `Run` rather than the constructor.  Can we do the same here and create these in `RunWithArgs`?

Unless there is a reason this has to be in the constructor I think we'd be better off keeping the classes consistent across platforms.

Reply:
Initializing these in the Constructor makes it easy for UT. Dependencies (`repoRegistry`, `requestHandler`) can be Mocked and the Mock objects can be injected as arguments to the Constructor. This is the pattern I use in the upcoming UT PR. We can probably use this pattern in `GVFSService.Windows.cs` if we decide to do UT for that class.

Reply:
> We can probably use this pattern in GVFSService.Windows.cs if we decide to do UT for that class.

Even if we're not writing UT for Windows in the near future, can we still make the code structure on Mac and Windows consistent?

Reply:
If you add a TODO to your PR/Work Item for your UT work I am okay with deferring the change until then, I just don't want us to lose track of it 😄

Reply:
Created a tracking [issue](https://github.com/Microsoft/VFSForGit/issues/1045)